MidnightBSD provides the FreeBSD process sysctl interface used by polkit.
Meson reports its system name as midnightbsd, so select the FreeBSD code path
instead of the Linux /proc parser.

--- meson.build.orig	2025-12-17 16:14:53 UTC
+++ meson.build
@@ -170,6 +170,9 @@ endif
 # check OS
 host_system = host_machine.system()
 config_data.set('HAVE_' + host_system.to_upper(), true)
+if host_system == 'midnightbsd'
+  config_data.set('HAVE_FREEBSD', true)
+endif
 
 # Check whether setnetgrent has a return value
 have_netgroup_h = cc.has_header('netgroup.h')
